test/common/ops/changeClass.js

```diff
@@ -27,6 +28,19 @@ describe('shared.ops.changeClass', () => {
     }
   });
 
+  it('req.query.class is an invalid class', (done) => {
+    user.flags.classSelected = false;
+    user.preferences.disableClasses = false;
+
+    try {
+      changeClass(user, {query: {class: 'cellist'}});
+    } catch (err) {
+      expect(err).to.be.an.instanceof(BadRequest);
+      expect(err.message).to.equal(i18n.t('invalidClass'));
+      done();
+    }
```

Comment:
Yes, I can confirm that if done isn't called in 8s the test will fail. It would probably be a little more efficient not to wait for this bug given that tests usually don't fail it's not a big deal if one day a build takes 8s more than usual